src/main/scala/gitbucket/core/service/ActivityService.scala

```diff
@@ -335,46 +335,8 @@ trait ActivityService {
     )
   }
 
-  def recordCreateWikiPageActivity(
-    userName: String,
-    repositoryName: String,
-    activityUserName: String,
-    pageName: String
-  ): Unit = {
-    writeLog(
-      Activity(
-        userName,
-        repositoryName,
-        activityUserName,
-        "create_wiki",
-        s"[user:${activityUserName}] created the [repo:${userName}/${repositoryName}] wiki",
-        Some(pageName),
-        currentDate,
-        UUID.randomUUID().toString
-      )
-    )
-  }
-
-  def recordEditWikiPageActivity(
-    userName: String,
-    repositoryName: String,
-    activityUserName: String,
-    pageName: String,
-    commitId: String
-  ): Unit = {
-    writeLog(
-      Activity(
-        userName,
-        repositoryName,
-        activityUserName,
-        "edit_wiki",
-        s"[user:${activityUserName}] edited the [repo:${userName}/${repositoryName}] wiki",
-        Some(pageName + ":" + commitId),
-        currentDate,
-        UUID.randomUUID().toString
-      )
-    )
-  }
+  def recordWikiActivity(maybeActivityInfo: Option[WikiActivityActionInfo]): Unit =
```

Comment:
Converting the two functions we had into this generic function which accepts a trait that has subtypes specific to each type of activity has negated the need for having a specific function for each activity.